Source/CarthageKit/Xcode.swift

```diff
@@ -6,6 +6,17 @@ import ReactiveSwift
 import ReactiveTask
 import XCDBLD
 
+/// Gets the shared cache file path URL for dependency at pinned version
+/// ~/Library/Caches/org.carthage.CarthageKit/SharedCache/XcodeVersion/DependencyName/Version, exemple :
+/// ~/Library/Caches/org.carthage.CarthageKit/SharedCache/9.4.1_9F2000/SwiftyUserDefaults/3.0.1
+internal func sharedCacheURLByXcodeVersionFor(dependency: Dependency, pinnedVersion: PinnedVersion) -> URL {
+	let sharedCacheURL = Constants.userCachesURL.appendingPathComponent("SharedCache")
+	let sharedCachePerXcode = sharedCacheURL.appendingPathComponent(XcodeVersion.makeString(), isDirectory: true)
```

Comment:
I think you'll want to use the swift toolchain version, not the xcode version. You could switch toolchains within xcode leading to incompatible builds.

Reply:
you're right, i've replaced Xcode version with the toolchain one if it is set (kept XcodeVersion if no toolchain set in BuildOptions)

Reply:
new version visible here : https://github.com/Carthage/Carthage/commit/23d433f835d985d34ee6d8589230e478416268a2#diff-f66644e8cfbcad3c3826395a690ddfbcR12

Reply:
Why not just use the xcrun swift --version like other places in the code? Even if not set explicitly (default) it will yield the correct swift toolchain version used.